app/src/main/java/org/wikipedia/views/SuggestedArticleDescriptionsReportDialog.kt

```diff
@@ -0,0 +1,68 @@
+package org.wikipedia.views
+
+import android.app.Activity
+import android.content.Context
+import android.graphics.Color
+import android.graphics.drawable.ColorDrawable
+import androidx.appcompat.app.AlertDialog
+import org.wikipedia.R
+import org.wikipedia.analytics.eventplatform.MachineGeneratedArticleDescriptionsAnalyticsHelper
+import org.wikipedia.databinding.DialogDescriptionSuggestionReportBinding
+import org.wikipedia.util.FeedbackUtil
+
+class SuggestedArticleDescriptionsReportDialog(context: Context, suggestion: String, callback: Callback) : AlertDialog(context) {
+
+    fun interface Callback {
+        fun onReportClick()
+    }
+
+    private var reported = false
+    private val binding = DialogDescriptionSuggestionReportBinding.inflate(layoutInflater)
+
+    init {
+        setView(binding.root)
+        window?.setBackgroundDrawable(ColorDrawable(Color.TRANSPARENT))
+        binding.reportButton.setOnClickListener {
+            collectReportData(suggestion)
+            FeedbackUtil.makeSnackbar(context as Activity,
+                context.getString(R.string.suggested_edits_suggestion_report_submitted)).show()
+            callback.onReportClick()
+            reported = true
+            dismiss()
+        }
+        binding.suggestionReportOther.setEndIconOnClickListener {
+            binding.suggestionReportOther.editText?.text?.clear()
+        }
+        binding.cancelButton.setOnClickListener { dismiss() }
+        setOnDismissListener {
+            if (!reported) {
+                MachineGeneratedArticleDescriptionsAnalyticsHelper.logReportDialogOptedOut(context,
+                    suggestion, getReportReasons())
+            }
+        }
+    }
+
+    private fun getReportReasons(): List<String> {
+        val responses = mutableListOf<String>()
+
+        if (binding.notEnoughInfoCheckbox.isChecked) {
+            responses.add(binding.notEnoughInfoCheckbox.text.toString())
```

Comment:
The "report reasons" should probably be the resource names of the buttons, not the localized text. (except the "other" reason)